server/raft_transport_test.go

```diff
@@ -142,3 +143,72 @@ func TestSendAndReceive(t *testing.T) {
 		}
 	}
 }
+
+// TestInOrderDelivery verifies that for a given pair of nodes, raft
+// messages are delivered in order. This test is unfortunately not
+// very sensitive; even when we were not guaranteeing in-order
```

Comment:
Yes. I initially tested this change with (non-committed) changes to the rpc server itself to introduce delays, but I didn't want to commit that kind of change to non-test code. But I've added a new commit that does something similar in the test-only channelServer.